src/lib/models/deployment.js

```diff
@@ -89,10 +119,12 @@ module.exports = class Deployment {
     })
 
     _.forEach([
-      ['spec.template.spec.containers[0].ports' , this.ports()]     ,
-      ['spec.replicas'                          , this.instances()]
+      ['spec.template.spec.containers[0].ports'   , this.ports()]        ,
+      ['spec.template.spec.containers[0].volumes' , this.volumeMounts()] ,
+      ['spec.template.spec.volumes'               , this.volumes()]      ,
+      ['spec.replicas'                            , this.instances()]
     ], ([key, val]) => {
-      _.set(this.release, key, val)
+      _.set(this.release, this.replaceKey(key), val)
```

Comment:
I'm not clear on what you're doing here with replaceKey. Why are you translating volumes to volumeMounts? Why not just set the key here directly?

Reply:
The keyframe volume syntax insn't translated 1-1 to kubernetes, but declared as an abstraction.
keyMappings allows renaming or even reusing abstractions in several elements (in this case Kubernetes requires declaring volumes as volumeMounts under container and as volumes under spec, so keyframe volumes may be used for both).

Reply:
Right, but why not just add that to the logic directly in the Deployment class? This is already k8s specific.

Reply:
Indeed overcomplicated this.  Looks much better now :). Thanks !